alphagov/paas-cf
# [#132651425] Upgrade to CF 245, Diego 0.1487.0, CAPI 1.8.0

[#132651425 Full fix for CVE 2016-6655: Utility script command injection (aka Upgrade to CF 245)](https://github.com/cloudfoundry/capi-release/releases/tag/1.8.0)

What?
-----

A critical script injection vulnerability has been discovered in a utility logging script, which forced us to quickly patch gorouter in #552

https://lists.cloudfoundry.org/archives/list/cf-dev@lists.cloudfoundry.org/thread/42YUJU2N27HBPFVMZR2QM7JI6YSEKORR/#42YUJU2N27HBPFVMZR2QM7JI6YSEKORR

But the full set of patches across all the services is in the [CF version v245](https://github.com/cloudfoundry/cf-release/tree/v245). We decided to upgrade to that version to get the right patch across.

The upgrade requires several manifest changes and different releases. We highlight:
 * [CF version v245](https://github.com/cloudfoundry/cf-release/releases/tag/v245)
 * [Diego version 0.1487.0](https://github.com/cloudfoundry/diego-release/releases/tag/v0.1487.0), as recommened
 * Adhoc [CAPI version 1.8.0](https://github.com/cloudfoundry/capi-release/releases/tag/1.8.0) instead of bundled version 1.7.0 in cf-release which includes a patched cloud_controller to fix a bug that prevented python apps to be deployed without `Procfile`\
 * etcd can be configured with TLS support, but we keep it disabled due the complexity of the upgrade procedure. It will be done in a different story.

How to test?
------------

All tests should pass. You should test:

 * Upgrade path:
   1. Deploy from master.
   2. Deploy from this branch

 * New installation: deploy directly from this branch.

Optionally, check that you can deploy a python app without Procfile, for instance 

https://github.com/ihuston/python-cf-examples/tree/master/01-simple-python-app

Who?
----

Anyone but @keymon or @alext
